.github/workflows/major-version-updater.yaml

```diff
@@ -0,0 +1,33 @@
+---
+name: "Major Version Updater"
+on:
+  workflow_call:
+    inputs:
+      tag_name:
+        required: true
+        type: string
+permissions:
+  contents: read
+jobs:
+  major_version_updater:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4.2.2
```

Comment:
Can we use sha instead of version here?

Reply:
checkout is one of our (GitHub) immutable actions.  [It's workflow](https://github.com/actions/checkout/blob/main/.github/workflows/publish-immutable-actions.yml).  

The list of immutable actions I'm aware of are the following:

```
"actions/checkout",
 "actions/setup-node",
 "actions/github-script",
 "actions/stale",
 "actions/download-artifact",
 "actions/upload-artifact",
 "actions/cache",
 "actions/labeler",
 "actions/setup-go",
 "actions/setup-python",
 "actions/create-github-app-token",
 "actions/deploy-pages",
 "actions/upload-pages-artifact",
 "actions/configure-pages",
 "octokit/request-actions",
 "actions/setup-java",
 "actions/setup-dotnet"
```

Reply:
I can switch to SHA if we really want that

Reply:
For the record, if it wasn't an immutable action, I would use a SHA and if I wasn't, it should be a blocker :smile:

Reply:
Good to go as is